test/e2e/cluster_upgrade_runtimesdk.go

```diff
@@ -184,7 +169,7 @@ func clusterUpgradeWithRuntimeSDKSpec(ctx context.Context, inputGetter func() cl
 			PreWaitForCluster: func() {
 				beforeClusterCreateTestHandler(ctx,
 					input.BootstrapClusterProxy.GetClient(),
-					namespace.Name, clusterName,
+					clusterRef,
```

Comment:
I wonder if we should just pass in a types.NamespacedName instead. If I saw correctly we only set Namespace+Name and we also only use those two fields. Feels a bit dangerous to hand over a full cluster while it is almost entirely empty and we only need Namespace+Name

Reply:
done